File: go/isogram/isogram.go
// Package isogram contains the solution for the Isogram exercise from Exercism.
package isogram

import "unicode"

// IsIsogram reports whether the text is an isogram.
func IsIsogram(text string) bool {
	encounteredRune := make(map[rune]bool)

	for _, r := range text {
		r = unicode.ToLower(r)

		if !unicode.IsLetter(r) {
			continue
		}

		if encounteredRune[r] {
			return false
		}

		encounteredRune[r] = true
	}

	return true
}

File: c/isogram/isogram.c
#include <ctype.h>
#include <stdint.h>

#include "isogram.h"

bool is_isogram(const char phrase[]) {
    if (!phrase) {
        return false;
    }
    
    uint32_t letters = 0;
    for (int i = 0; phrase[i]; i++) {
        int ch = tolower(phrase[i]);
        if (ch < 'a' || ch > 'z') {
            continue;
        }
        
        if ((letters & 1 << (ch - 'a')) > 0) {
            return false;
        }

        letters |= 1 << (ch - 'a');
    }

    return true;
}